### ISSUE TITLE
rancher2_bootstrap is failing on first tf run

### ISSUE THREAD

**Issue by mikekuzak** [rancher/terraform-provider-rancher2#493]
Hi 

terrafrom 0.13.5

I'm using the docker provisoner to setup rancher, bootstrap and create a custom cluster.
```
# ------------------------------------------------------------------------------
# CREATE A RANDOM PASSWORD
# ------------------------------------------------------------------------------

resource "random_password" "password" {
  length           = 10
  special          = true
  override_special = "_%@"
}

# ------------------------------------------------------------------------------
# LOCAL VARIABLES
# ------------------------------------------------------------------------------

locals {
  rancher_console_pw = random_password.password.result
  env                = terraform.workspace
}

# ------------------------------------------------------------------------------
# SET PROVIDERS
# ------------------------------------------------------------------------------

provider "docker" {
  host = "ssh://${var.rancher_host_ssh}:22"
}

provider "rancher2" {
  alias     = "bootstrap"
  api_url   = "https://${var.rancher_api_url}"
  bootstrap = true
  insecure  = true
  retries   = 100
}

resource "rancher2_bootstrap" "admin" {
  depends_on = [module.docker_rancher]
  provider   = rancher2.bootstrap

  password  = local.rancher_console_pw
  telemetry = false
}

provider rancher2 {
  alias = "admin"

  api_url   = rancher2_bootstrap.admin.url
  token_key = rancher2_bootstrap.admin.token
  insecure  = true
}
```

This seems to fail on the first run, but is good on the second run. 

```
Initializing provider plugins...
- Using previously-installed terraform-providers/docker v2.7.2
- Using previously-installed hashicorp/null v3.0.0
- Using previously-installed hashicorp/local v2.0.0
- Using previously-installed rancher/rancher2 v1.10.4
- Using previously-installed hashicorp/random v3.0.0
- Using previously-installed hashicorp/kubernetes v1.13.3

random_password.password: Creating...
random_password.password: Creation complete after 0s [id=none]
module.docker_rancher.docker_image.image: Creating...
module.docker_rancher.docker_image.image: Still creating... [10s elapsed]
module.docker_rancher.docker_image.image: Still creating... [20s elapsed]
module.docker_rancher.docker_image.image: Still creating... [30s elapsed]
module.docker_rancher.docker_image.image: Still creating... [40s elapsed]
module.docker_rancher.docker_image.image: Creation complete after 45s [id=sha256:2159e92d1f8d90cce01eb0b7e533023e9c0b2d77dd366030af84741009afb374rancher/rancher:v2.5.1]
module.docker_rancher.docker_container.container: Creating...
module.docker_rancher.docker_container.container: Creation complete after 2s [id=f21b785556a0b1106d127031ee775ace2c680f1dc68ea278bdf93a25f6109162]
rancher2_bootstrap.admin: Creating...
rancher2_bootstrap.admin: Still creating... [10s elapsed]
rancher2_bootstrap.admin: Still creating... [20s elapsed]
rancher2_bootstrap.admin: Still creating... [30s elapsed]
rancher2_bootstrap.admin: Still creating... [40s elapsed]
rancher2_bootstrap.admin: Still creating... [50s elapsed]
rancher2_bootstrap.admin: Still creating... [1m0s elapsed]

Error: [ERROR] Connecting with user/pass: Bad response statusCode [403]. Status [403 Forbidden]. Body: [baseType=error, code=Forbidden, message=settings.management.cattle.io "k8s-version" is forbidden: User "user-r5tbb" cannot get resource "settings" in API group "management.cattle.io" at the cluster scope] from [https://sbox-k8s:9443/v3/settings/k8s-version]

  on main.tf line 36, in resource "rancher2_bootstrap" "admin":
  36: resource "rancher2_bootstrap" "admin" {
```

I'm stuck on this at the moment. If I run it the second time it works. 
Any advise pls ? 
Thanks

**Comment by rawmind0** [comment]
This is issue may be related to https://github.com/rancher/terraform-provider-rancher2/issues/477

**Comment by mikekuzak** [comment]
@rawmind0 As a temp. workaround I added a delay and it seems to work.

```
resource "null_resource" "delay_docker_start" {
  provisioner "local-exec" {
    command = "sleep 90"
  }
}

resource "rancher2_bootstrap" "admin" {
  depends_on = [module.docker_rancher, null_resource.delay_docker_start]
  provider   = rancher2.bootstrap

  password  = local.rancher_console_pw
  telemetry = false
}
```

```
null_resource.delay_docker_start: Creating...
null_resource.delay_docker_start: Provisioning with 'local-exec'...
null_resource.delay_docker_start (local-exec): Executing: ["/bin/sh" "-c" "sleep 90"]
random_password.password: Creating...
random_password.password: Creation complete after 0s [id=none]
module.docker_rancher.docker_image.image: Creating...
module.docker_rancher.docker_image.image: Creation complete after 0s [id=sha256:2159e92d1f8d90cce01eb0b7e533023e9c0b2d77dd366030af84741009afb374rancher/rancher:v2.5.1]
module.docker_rancher.docker_container.container: Creating...
module.docker_rancher.docker_container.container: Creation complete after 2s [id=a790416138d20bae79ea92e33f41fab094a2e77bb4177b7d58a51e44f31935ca]
null_resource.delay_docker_start: Still creating... [10s elapsed]
null_resource.delay_docker_start: Still creating... [20s elapsed]
null_resource.delay_docker_start: Still creating... [30s elapsed]
null_resource.delay_docker_start: Still creating... [40s elapsed]
null_resource.delay_docker_start: Still creating... [50s elapsed]
null_resource.delay_docker_start: Still creating... [1m0s elapsed]
null_resource.delay_docker_start: Still creating... [1m10s elapsed]
null_resource.delay_docker_start: Still creating... [1m20s elapsed]
null_resource.delay_docker_start: Still creating... [1m30s elapsed]
null_resource.delay_docker_start: Creation complete after 1m30s [id=788604808180930446]
rancher2_bootstrap.admin: Creating...
rancher2_bootstrap.admin: Still creating... [10s elapsed]
rancher2_bootstrap.admin: Creation complete after 19s [id=user-7sll7]
```

**Comment by rawmind0** [comment]
@mikekuzak , your issue seems related to https://github.com/rancher/terraform-provider-rancher2/issues/477 .  Best solution is sync your module, not starting rancher2 provider until Rancher is up and running. As an option, you can try with provider argument [retries](https://registry.terraform.io/providers/rancher/rancher2/latest/docs#retries). https://github.com/rancher/terraform-provider-rancher2/issues/477#issuecomment-707680326

**Comment by mikekuzak** [comment]
> @mikekuzak , your issue seems related to #477 . Best solution is sync your module, not starting rancher2 provider until Rancher is up and running. As an option, you can try with provider argument [retries](https://registry.terraform.io/providers/rancher/rancher2/latest/docs#retries). [#477 (comment)](https://github.com/rancher/terraform-provider-rancher2/issues/477#issuecomment-707680326)

I already have the reties, didn't help. 

provider "rancher2" {
  alias     = "bootstrap"
  api_url   = "https://${var.rancher_api_url}"
  bootstrap = true
  insecure  = true
  retries   = 100
}

**Comment by rawmind0** [comment]
As mentioned, best option is to sync tf modules. Anyway, added a fix on PR #498 to configure number of user/pass login retries on bootstrap, based on `retries` provider argument

**Comment by empeekdev** [comment]
@rawmind0 We have the same issue on rancher 2.6.3, 2.6.8. But in our case this problem happens even after 10 tries. We just cannot bootstrap rancher, don’t know what to do. Any ideas?
